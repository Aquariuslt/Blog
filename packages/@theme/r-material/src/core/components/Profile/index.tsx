import * as React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import Grid from '@material-ui/core/Grid';

interface ProfileProps {
  name: string;
  avatar: string;
  subtitle: string;
  description: string;
}

const useStyles = makeStyles({
  avatar: {
    margin: 10,
    width: 60,
    height: 60
  }
});

const Profile: React.ComponentType<ProfileProps> = (props: ProfileProps): JSX.Element => {
  const classes = useStyles();

  return (
    <Grid container justify="center" alignItems="center">
      <Avatar component="div" alt={props.name} src={props.avatar} className={classes.avatar} />
    </Grid>
  );
};

export default Profile;
