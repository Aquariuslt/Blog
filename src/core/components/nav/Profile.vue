<template>
  <div>
    <md-card class="md-primary">
      <md-card-header>
        <md-avatar>
          <img :src="avator" alt="avator">
        </md-avatar>
        <div class="md-title">{{author}}</div>
        <div class="md-subhead">{{description}}</div>
      </md-card-header>
    </md-card>
  </div>
</template>

<script>
  export default {
    name: 'profile',
    props: {
      author: String,
      avator: String,
      description: String
    }
  };
</script>

<style scoped>

</style>
