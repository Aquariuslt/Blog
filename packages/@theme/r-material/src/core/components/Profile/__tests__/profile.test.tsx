import * as React from 'react';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

import Profile from '../';

Enzyme.configure({ adapter: new Adapter() });

describe('@theme/r-material: core/components: Profile', (): void => {
  it('# should mount profile component correctly', (): void => {
    const mockProfileProps = {
      name: '',
      avatar: 'https://img.aquariuslt.com/social/avatar.png',
      subtitle: '',
      description: ''
    };

    const wrapper = Enzyme.shallow(<Profile {...mockProfileProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
