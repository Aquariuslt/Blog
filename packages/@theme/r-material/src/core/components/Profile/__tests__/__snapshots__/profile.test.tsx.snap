// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@theme/r-material: core/components: Profile # should mount profile component correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="makeStyles-root-1"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Avatar))
    alt=""
    className="makeStyles-avatar-2"
    component="div"
    src="https://img.aquariuslt.com/social/avatar.png"
  />
</WithStyles(ForwardRef(Grid))>
`;
